### ISSUE TITLE
Build error in Ubuntu 16.04

### ISSUE THREAD

**Issue by echotor** [Microsoft/cpprestsdk#603]
When I follow this `https://github.com/Microsoft/cpprestsdk/wiki/How-to-build-for-Linux` to build cpprestsdk, I can not build successfully. The error is like:
```
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Setting gcc options
CMake Error at src/CMakeLists.txt:14 (list):
  list does not recognize sub-command FILTER


-- websocketpp not found, using the embedded version
-- Boost version: 1.58.0
-- Found the following Boost libraries:
--   random
--   system
--   thread
--   filesystem
--   chrono
--   atomic
--   date_time
--   regex
-- Found OpenSSL: /usr/lib/x86_64-linux-gnu/libssl.so;/usr/lib/x86_64-linux-gnu/libcrypto.so (found suitable version "1.0.2g", minimum required is "1.0.0") 
-- Performing Test _SSL_LEAK_SUPPRESS_AVAILABLE
-- Performing Test _SSL_LEAK_SUPPRESS_AVAILABLE - Success
-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version "1.2.8") 
-- Added test library httpclient_test
-- Added test library httplistener_test
-- Added test library json_test
-- Added test library pplx_test
-- Added test library streams_test
-- Added test library uri_test
-- Added test library utils_test
-- Added test library websocketsclient_test
-- Configuring incomplete, errors occurred!
See also "/home/wangyawei/Public/cpprestsdk/Release/build/CMakeFiles/CMakeOutput.log".
See also "/home/wangyawei/Public/cpprestsdk/Release/build/CMakeFiles/CMakeError.log".
```
and the log text is below:
```
Determining if the pthread_create exist failed with the following output:
Change Dir: /home/wangyawei/Public/cpprestsdk/Release/build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/make" "cmTC_687eb/fast"
/usr/bin/make -f CMakeFiles/cmTC_687eb.dir/build.make CMakeFiles/cmTC_687eb.dir/build
make[1]: Entering directory '/home/wangyawei/Public/cpprestsdk/Release/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_687eb.dir/CheckSymbolExists.c.o
/usr/bin/cc     -o CMakeFiles/cmTC_687eb.dir/CheckSymbolExists.c.o   -c /home/wangyawei/Public/cpprestsdk/Release/build/CMakeFiles/CMakeTmp/CheckSymbolExists.c
Linking C executable cmTC_687eb
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_687eb.dir/link.txt --verbose=1
/usr/bin/cc       CMakeFiles/cmTC_687eb.dir/CheckSymbolExists.c.o  -o cmTC_687eb -rdynamic 
CMakeFiles/cmTC_687eb.dir/CheckSymbolExists.c.o: In function `main':
CheckSymbolExists.c:(.text+0x16): undefined reference to `pthread_create'
collect2: error: ld returned 1 exit status
CMakeFiles/cmTC_687eb.dir/build.make:97: recipe for target 'cmTC_687eb' failed
make[1]: *** [cmTC_687eb] Error 1
make[1]: Leaving directory '/home/wangyawei/Public/cpprestsdk/Release/build/CMakeFiles/CMakeTmp'
Makefile:126: recipe for target 'cmTC_687eb/fast' failed
make: *** [cmTC_687eb/fast] Error 2

File /home/wangyawei/Public/cpprestsdk/Release/build/CMakeFiles/CMakeTmp/CheckSymbolExists.c:
/* */
#include <pthread.h>

int main(int argc, char** argv)
{
  (void)argv;
#ifndef pthread_create
  return ((int*)(&pthread_create))[argc];
#else
  (void)argc;
  return 0;
#endif
}

Determining if the function pthread_create exists in the pthreads failed with the following output:
Change Dir: /home/wangyawei/Public/cpprestsdk/Release/build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/make" "cmTC_2a3f4/fast"
/usr/bin/make -f CMakeFiles/cmTC_2a3f4.dir/build.make CMakeFiles/cmTC_2a3f4.dir/build
make[1]: Entering directory '/home/wangyawei/Public/cpprestsdk/Release/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_2a3f4.dir/CheckFunctionExists.c.o
/usr/bin/cc    -DCHECK_FUNCTION_EXISTS=pthread_create   -o CMakeFiles/cmTC_2a3f4.dir/CheckFunctionExists.c.o   -c /usr/share/cmake-3.5/Modules/CheckFunctionExists.c
Linking C executable cmTC_2a3f4
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_2a3f4.dir/link.txt --verbose=1
/usr/bin/cc   -DCHECK_FUNCTION_EXISTS=pthread_create    CMakeFiles/cmTC_2a3f4.dir/CheckFunctionExists.c.o  -o cmTC_2a3f4 -rdynamic -lpthreads 
/usr/bin/ld: cannot find -lpthreads
collect2: error: ld returned 1 exit status
CMakeFiles/cmTC_2a3f4.dir/build.make:97: recipe for target 'cmTC_2a3f4' failed
make[1]: *** [cmTC_2a3f4] Error 1
make[1]: Leaving directory '/home/wangyawei/Public/cpprestsdk/Release/build/CMakeFiles/CMakeTmp'
Makefile:126: recipe for target 'cmTC_2a3f4/fast' failed
make: *** [cmTC_2a3f4/fast] Error 2
```

Could someone tell me about this error?

**Comment by ras0219-msft** [comment]
Thank you for reporting this error.

The first issue was caused by the use of a function added in CMake 3.6; I've pushed the fix to master, so could you please try again with the latest master and let me know if it works?

**Comment by echotor** [comment]
OK, I try it and it complete.
```
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Setting gcc options
-- Found websocketpp version 0.7.0 on system
-- Boost version: 1.58.0
-- Found the following Boost libraries:
--   random
--   system
--   thread
--   filesystem
--   chrono
--   atomic
--   date_time
--   regex
-- Found OpenSSL: /usr/lib/x86_64-linux-gnu/libssl.so;/usr/lib/x86_64-linux-gnu/libcrypto.so (found suitable version "1.0.2g", minimum required is "1.0.0") 
-- Performing Test _SSL_LEAK_SUPPRESS_AVAILABLE
-- Performing Test _SSL_LEAK_SUPPRESS_AVAILABLE - Success
-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version "1.2.8") 
-- Added test library httpclient_test
-- Added test library httplistener_test
-- Added test library json_test
-- Added test library pplx_test
-- Added test library streams_test
-- Added test library uri_test
-- Added test library utils_test
-- Added test library websocketsclient_test
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:
    CMAKE_BUILD_TYPR


-- Build files have been written to: /home/wangyawei/Public/cpprestsdk/Release/build
```

**Comment by SergeevGregory** [comment]
Dear Developers,
I try to build cpprestsdk on my Ubuntu 16.04.4, but I have issue with lpthreads.
Can you help me with this? Thx.

**Comment by echotor** [comment]
@SergeevGregory check the link https://github.com/Microsoft/cpprestsdk/issues/600

**Comment by SergeevGregory** [comment]
@echotor unfortunately I can't understand how you fix in in [#600](https://github.com/Microsoft/cpprestsdk/issues/600). Can you help me? Please, explain your fix step by step :)
Thx.
